### PR TITLE
page-object-model

### PR DESCRIPTION
**Pull request by JuanManu59**

### CODE REVIEW

**Comment by JuanManu59** [comment]
i1
i2

**Comment by JuanManu59** [comment]
image

No comprendo por que Travis CI - Branch falló, me pueden comentar por que creen que sea, Gracias!!.

File: src/page/shipping-step.page.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { $, ElementFinder } from 'protractor';
+
+export class ShippingStepPage {
```
Comment (holgiosalos) [inline src/page/shipping-step.page.ts:3]: this class should have two steps, first the acceptance of the terms of service (`'#cgv'`) and then the click on the "Proceed to checkout" button (`'#form > p > button > span'`)

File: src/page/sign-in-step.page.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import { $, ElementFinder } from 'protractor';
+
+export class SignInStepPage {
```
Comment (holgiosalos) [inline src/page/sign-in-step.page.ts:3]: In this PageObject the idea is to encapsulate the steps required to make the sign in. You only have the click on the login button, please include the steps to write email and password.

File: test/buy-tshirt.spec.ts [hunk 3]
```diff
@@ -1,38 +1,80 @@
...
+
     await(browser.sleep(3000));
-    await $('.cart_navigation span').click();
+
+    // await $('.cart_navigation span').click();
+    await summaryStepPage.method2();
+
     await(browser.sleep(3000));
 
     await $('#email').sendKeys('aperdomobo@gmail.com');
```
Comment (holgiosalos) [inline test/buy-tshirt.spec.ts]: email and password actions should go on SignInStepPage

File: test/buy-tshirt.spec.ts [hunk 4]
```diff
@@ -1,38 +1,80 @@
...
 
-    await $('#cgv').click();
+    // await $('#cgv').click();
+    await shippingStepPage.method5();
+
     await(browser.sleep(3000));
 
-    await $('#form > p > button > span').click();
+    // await $('#form > p > button > span').click();
+    await paymentStepPage.method6();
```
Comment (holgiosalos) [inline test/buy-tshirt.spec.ts]: this step should go on the ShippingStepPage

File: test/buy-tshirt.spec.ts [hunk 5]
```diff
@@ -1,38 +1,80 @@
...
+
     await(browser.sleep(3000));
-    await $('#HOOK_PAYMENT > div:nth-child(1) > div > p > a').click();
+
+    // await $('#HOOK_PAYMENT > div:nth-child(1) > div > p > a').click();
+    await bankPaymentPage.method7();
```
Comment (holgiosalos) [inline test/buy-tshirt.spec.ts]: This step belongs to the PaymentStepPage.

File: test/buy-tshirt.spec.ts [hunk 6]
```diff
@@ -1,38 +1,80 @@
...
+
     await(browser.sleep(3000));
-    await $('#cart_navigation > button > span').click();
+
+    // await $('#cart_navigation > button > span').click();
+    await orderSummaryPage.method8();
```
Comment (holgiosalos) [inline test/buy-tshirt.spec.ts]: This step should go on BankPaymentPage

File: test/buy-tshirt.spec.ts [hunk 7]
```diff
@@ -1,38 +1,80 @@
...
+
     await(browser.sleep(3000));
 
     await expect($('#center_column > div > p > strong').getText())
```
Comment (holgiosalos) [inline test/buy-tshirt.spec.ts]: The action of getting the confirmation text should go on the OrderSummaryPage

**Review by holgiosalos — COMMENTED** [review]
According to what we talked about in the zoom meeting, change the method names as well as their related attributes, to describe the element and the action you are interacting on each page.

File: src/page/sign-in-step.page.ts [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+  private credentials: ElementFinder;
+
+  constructor () {
+    this.credentials = $('#SubmitLogin > span');
+  }
+
+  public async goToSignIn(): Promise<void> {
+    await $('#email').sendKeys('aperdomobo@gmail.com');
+    await $('#passwd').sendKeys('WorkshopProtractor');
```
Comment (yesid-lopez) [inline src/page/sign-in-step.page.ts]: move as method params instead of hardcoding the user and the password

File: test/buy-tshirt.spec.ts [hunk 9]
```diff
@@ -1,41 +1,77 @@
...
+    // await $('#block_top_menu > ul > li:nth-child(3) > a').click(); CHECK
+    // await $('#center_column a.button.ajax_add_to_cart_button.btn.btn-default').click(); CHECK
+    // await $('[style*="display: block;"] .button-container > a').click(); CHECK
+    // await $('.cart_navigation span').click(); CHECK
+    // await $('#SubmitLogin > span').click(); CHECK
+    // await $('#center_column > form > p > button > span').click(); CHECK
+    // await $('#cgv').click(); CHECK
+    // await $('#form > p > button > span').click(); CHECK
+    // await $('#HOOK_PAYMENT > div:nth-child(1) > div > p > a').click(); CHECK
+    // await $('#cart_navigation > button > span').click();
```
Comment (yesid-lopez) [inline test/buy-tshirt.spec.ts:38]: delete comments

File: src/page/order-summary.page.ts [hunk 10]
```diff
@@ -0,0 +1,12 @@
+import { $ } from 'protractor';
+
+export class OrderSummaryPage {
+
+  constructor () {
+  }
+
+  public async goToOrder(): Promise<void> {
+    await expect($('#center_column > div > p > strong').getText())
```
Comment (holgiosalos) [inline src/page/order-summary.page.ts]: The idea with POM is just encapsulating business behaviors; verifications or expects should be done just in test classes. Please create an attribute for the confirmation text ant make this function return that value

File: src/page/sign-in-step.page.ts (hunk 8 above)
Comment (holgiosalos) [reply src/page/sign-in-step.page.ts]: Agree, please add two params to this method, one for email and the other one for password, and from the test class, you can pass the values. In this way, this page object will not be coupled to a specific user and password

**Comment by JuanManu59** [comment]
i2
i1

**Review by holgiosalos — APPROVED** [review]

**Pull request merged**